### PR TITLE
Enable inference mode to reduce VRAM usage during preprocessing

### PR DESCRIPTION
**Pull request by akutzer**
Enable `torch.inference_mode()` during preprocessing so no gradients, etc. are computed during feature extraction

### CODE REVIEW

File: stamp/preprocessing/helpers/feature_extractors.py [hunk 1]
```diff
@@ -153,9 +153,11 @@ def extract_features_(
     dtype = next(model.parameters()).dtype
 
     feats = []
-    for batch in tqdm(dl, leave=False):
-        feats.append(
-            model(batch.type(dtype).to(device)).half().cpu().detach())
+    with torch.inference_mode():
```
Comment (akutzer) [inline stamp/preprocessing/helpers/feature_extractors.py:156]: One could also think about also enabling mixed precission like:
```python
with torch.inference_mode(), torch.autocast(self.device.type):
```
